### PR TITLE
Replace `if ... in spec` with `spec.satisfies` in f* and g* packages

### PR DESCRIPTION
**Pull request by aurianer**
with the structure `if "+variant" in spec:`, checks like `if "mpi" in spec:` would be valid whereas it could be `~mpi` or `+mpi`. Satisfies adds additional checks to avoid mistakes from the user.

### CODE REVIEW

**Review by rbberger — APPROVED** [review]
Remind me, why are we moving away from the `in spec` syntax?

**Comment by aurianer** [comment]
https://github.com/spack/spack/pull/44774#discussion_r1647649109

**Review by bonachea — APPROVED** [review]
I don't object to the change, but please update the PR and commit description with a comment explaining _why_ this change is motivated.

File: var/spack/repos/builtin/packages/flamemaster/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/flamemaster/package.py]:
```suggestion
        if spec.satisfies("^[virtuals=lapack] amdlibflame"):
```
@amd-toolchain-support Please verify this addition of explicit virtuals for `flamemaster ^amdlibflame`, per https://spack.readthedocs.io/en/latest/basic_usage.html#explicit-binding-of-virtual-dependencies.

File: var/spack/repos/builtin/packages/gromacs/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/gromacs/package.py]:
```suggestion
            if self.spec.satisfies("^[virtuals=fftw-api] amdfftw"):
```
@mabraham @eirrgang @junghans Please verify this addition of explicit virtuals for `gromacs ^amdfftw`, per https://spack.readthedocs.io/en/latest/basic_usage.html#explicit-binding-of-virtual-dependencies.

**Review by wdconinc — COMMENTED** [review]
Two instances of a virtual package dependency that should probably define through which virtual dependency the specific dependency is pulled in.

Verified all others and no issues.

Agreed with @bonachea that a better description should be added.

**Comment by aurianer** [comment]
> I don't object to the change, but please update the PR and commit description with a comment explaining _why_ this change is motivated.

Let me know if the PR description now is enough

File: var/spack/repos/builtin/packages/gromacs/package.py
Comment (mabraham) [reply var/spack/repos/builtin/packages/gromacs/package.py]: That looks good.

**Review by mabraham — APPROVED** [review]

File: var/spack/repos/builtin/packages/ginkgo/package.py
Comment (MarcelKoch) [inline var/spack/repos/builtin/packages/ginkgo/package.py:1]: No objections to the Ginkgo changes

**Review by MarcelKoch — APPROVED** [review]

**Review by luszczek — DISMISSED** [review]
(My personal view: operators are usually syntactic sugar and explicit methods are more Pythonic.)

**Comment by wdconinc** [comment]
Rebasing to pick up CI fix in https://github.com/spack/spack/commit/1c1970e7277440069c0050ebaf6720f72dcdf2e5

File: var/spack/repos/builtin/packages/gromacs/package.py
Comment (mabraham) [reply var/spack/repos/builtin/packages/gromacs/package.py]: I don't have a button to accept the suggested change, however

**Review by mabraham — APPROVED** [review]

File: var/spack/repos/builtin/packages/flamemaster/package.py
Comment (aurianer) [reply var/spack/repos/builtin/packages/flamemaster/package.py]: I updated it to use `virtuals`. In case, this can still be reverted

**Comment by aurianer** [comment]
@wdconinc this is good to go on my side, let me know if I should change anything still

**Review by wdconinc — APPROVED** [review]
Thanks for these changes. I'll approve and give it another day or so before merging in case of last minute comments.

**Pull request merged**
